src/pages/Address/index.tsx

```diff
@@ -19,39 +25,68 @@ export const Address = () => {
   const { address } = useParams<{ address: string }>()
   const { t } = useTranslation()
   const { currentPage, pageSize } = usePaginationParamsInListPage()
+  const { tx_status: txStatus } = useSearchParams('tx_status')
 
   // REFACTOR: avoid using useSortParam
   const { sortBy, orderBy, sort } = useSortParam<'time'>(s => s === 'time')
 
+  const isPendingTxListActive = txStatus === 'pending'
+
   const addressInfoQuery = useQuery(['address_info', address], () => explorerService.api.fetchAddressInfo(address))
 
-  const addressTransactionsQuery = useQuery(
-    ['address_transactions', address, currentPage, pageSize, sort],
-    async () => {
-      try {
-        const { data: transactions, total } = await explorerService.api.fetchTransactionsByAddress(
-          address,
-          currentPage,
-          pageSize,
-          sort,
-        )
+  const listQueryKey = [
+    isPendingTxListActive ? 'address_pending_transactions' : 'address_transactions',
+    address,
+    currentPage,
+    pageSize,
+    sort,
+  ]
+  const listQueryIns = isPendingTxListActive
+    ? explorerService.api.fetchPendingTransactionsByAddress
+    : explorerService.api.fetchTransactionsByAddress
+
+  const addressTransactionsQuery = useQuery(listQueryKey, async () => {
+    try {
+      const { data: transactions, total } = await listQueryIns(address, currentPage, pageSize, sort)
+      return {
+        transactions,
+        total,
+      }
+    } catch (err) {
+      const isEmptyAddress = isAxiosError(err) && err.response?.status === 404
+      if (isEmptyAddress) {
         return {
-          transactions,
-          total,
+          transactions: [],
+          total: 0,
         }
+      }
+      throw err
+    }
+  })
+
+  const pendingTransactionCountQuery = useQuery(
+    ['address_pending_transactions', address],
+    async () => {
+      try {
+        const { total } = await explorerService.api.fetchPendingTransactionsByAddress(address, 1, 10)
+        return total
       } catch (err) {
-        const isEmptyAddress = isAxiosError(err) && err.response?.status === 404
-        if (isEmptyAddress) {
-          return {
-            transactions: [],
-            total: 0,
-          }
-        }
-        throw err
+        return '-'
       }
     },
+    {
+      initialData: '-',
+    },
   )
 
+  const transactionCounts: Record<'committed' | 'pending', number | '-'> = {
+    committed:
+      addressInfoQuery.isFetched && addressInfoQuery.data
+        ? Number(addressInfoQuery.data.transactionsCount) ?? '-'
```

Comment:
I've checked that `transactionsCount` is defined as a `number` in the frontend type definition.

We can handle the conversion from the external data structure to the internal defined structure within the `service` layer.

If we don't do this, we would need to change the type definition to the correct `string` type.

Reply:
The handling here should be the same as for the Transaction below.

Add a string type to transactionsCount and provide a comment explaining the source of the string type.

Reply:
Comment was added by https://github.com/Magickbase/ckb-explorer-frontend/pull/175/commits/d56acfb29a1d877a45d95d39dea53bc1a7a4e91f